### PR TITLE
COMMERCE-12540

### PR DESCRIPTION
**Pull request by brianikim**
Hello @andrea-ale-sbarra ,

PR is mostly complete. Please take a look and let me know what you think.

We probably need to do additional review with @smottal to go over CSS and make sure my JS implementation is okay.

I am most likely going to need help from Stefano in resolving the edge case you see in the following picture in regards to multiple selects where the values are being returned as combined.

ExampleUI

https://liferay.atlassian.net/browse/COMMERCE-12540

cc/ @smottal .

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionNumeric.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ const ProductOptionNumeric = ({
 			);
 
 			if (option) {
-				[value] = option.value;
+				value = option.value;
```
Comment (smottal) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionNumeric.js:61]: Same here: is it correct we are removing the array?

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionRadio.js [hunk 2]
```diff
@@ -87,8 +87,10 @@ const ProductOptionRadio = ({
 							quantity: defaultProductOptionValue?.quantity,
 							skuId: defaultProductOptionValue?.skuId,
 							skuOptionKey: productOption.key,
+							skuOptionName: productOption.name,
 							skuOptionValueKey: defaultProductOptionValue?.key,
-							value: [defaultProductOptionValue?.key],
+							skuOptionValueName: defaultProductOptionValue?.name,
+							value: defaultProductOptionValue?.key,
```
Comment (smottal) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionRadio.js:93]: Same here: why not array? My thought is that is the service looking for arrays? Is it possible?

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionText.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ const ProductOptionText = ({
 			);
 
 			if (option) {
-				[value] = option.value;
+				value = option.value;
```
Comment (smottal) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionText.js:61]: Same here for array

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionDate.js [hunk 4]
```diff
@@ -58,7 +58,7 @@ const ProductOptionDate = ({
 			);
 
 			if (option) {
-				[value] = option.value;
+				value = option.value;
```
Comment (smottal) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionDate.js:61]: Same for array

**Review by smottal — CHANGES_REQUESTED** [review]
Hi @brianikim, could you please have a look to my notes? Thanks

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionCheckbox.js [hunk 5]
```diff
@@ -60,7 +60,7 @@ const ProductOptionCheckbox = ({
 
 			if (option) {
 				checked = option.value.includes(option.key);
-				[value] = option.value;
+				value = option.value;
```
[the remark this answers is not in the document]
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionCheckbox.js:63]: @smottal , the value is always be a single value, only `ProductOptionMultipleCheckbox` supports multiple values. I am removing the value as array for consistency purposes along with the other changes I made for the other fields such as numeric and date.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionCheckbox.js [hunk 6]
```diff
@@ -136,7 +138,8 @@ const ProductOptionCheckbox = ({
 				{
 					key: productOption.key,
 					skuOptionKey: productOption.key,
-					value: checked ? [value] : [],
+					skuOptionName: productOption.name,
+					value: checked ? value : '',
```
[the remark this answers is not in the document]
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionCheckbox.js:142]: We had a conversation about this @smottal . 
ProductOptionCheckbox is missing the implementations and we agreed that we can still support `skuOptionName` for now.
The values are never true/false. It is always the optionValueKey.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionNumeric.js (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionNumeric.js:61]: the value is always be a single value, only ProductOptionMultipleCheckbox supports multiple values. I am removing the value as array for consistency purposes along with the other changes I made for the other fields such as numeric and date.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionRadio.js (hunk 2 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionRadio.js:93]: The service is not looking for arrays. In the JS, these implementations are strictly for front-end purposes.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemViews/ItemInfoView.js [hunk 7]
```diff
@@ -41,17 +33,49 @@ function ItemInfoViewBundle({childItems}) {
...
+										<span>
+											{skuOptionValueName || value}
+										</span>
+
+										<span className="pl-2">
+											({quantity} &times; {name})
+										</span>
+
+										<span className="pl-2">
+											{skuUnitOfMeasure?.key || ''}
```
[the remark this answers is not in the document]
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemViews/ItemInfoView.js:67]: it is related it should go in parentesis

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionCheckboxMultiple.js [hunk 8]
```diff
@@ -149,12 +154,29 @@ const ProductOptionCheckboxMultiple = ({
...
+							? [
+									...currentSkuOptions[curSkuOptionIndex]
+										.skuOptionValueName,
+									curProductOptionValue.name,
+							  ]
+							: currentSkuOptions[
+									curSkuOptionIndex
+							  ].skuOptionValueName.filter(
+									(curVal) =>
+										!(curVal === curProductOptionValue.name)
```
[the remark this answers is not in the document]
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionCheckboxMultiple.js:178]: For this one, I just reused the same implementation we have for value and adapted it for `skuOptionValueName`. Shouldn't we be doing strict checking for this? In theory, both values being compared should be primitive string.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionRadio.js (hunk 2 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionRadio.js:93]: Please correct me here if I am wrong and I will re-fix accordingly.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemViews/ItemInfoView.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemViews/ItemInfoView.js]: Fixed in the latest commit.

[the remark this answers is not in the document]
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemViews/ItemInfoView.js]: <span className="pl-2">
									({quantity} &times; {name}{' '}
									{skuUnitOfMeasure?.key || ''})
								</span>

**Comment by brianikim** [comment]
@andrea-ale-sbarra , @smottal , I fixed some of the requests, but also requested clarifications/explanations on the other.
Please take a look and let me know.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionCheckbox.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (smottal) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionCheckbox.js:142]: Yep, I forgot it

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemViews/ItemInfoView.js [hunk 9]
```diff
@@ -41,17 +33,49 @@ function ItemInfoViewBundle({childItems}) {
 		>
 			<ClayPanel.Body>
 				<div className="child-items">
```
[the remark this answers is not in the document]
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemViews/ItemInfoView.js]: Resolved in new PR.

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemViews/ItemInfoView.js [hunk 10]
```diff
@@ -41,17 +33,49 @@ function ItemInfoViewBundle({childItems}) {
...
+										</span>
+
+										<span className="pl-2">
+											{skuUnitOfMeasure?.key || ''}
+										</span>
+									</span>
+								</p>
+							</div>
+						) : (
+							<div className="item-info-extra pt-2">
```
[the remark this answers is not in the document]
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemViews/ItemInfoView.js]: Agrees through slack discussion, removing the duplication by making a new const is fine.

**Comment by brianikim** [comment]
Reassigned for review. @andrea-ale-sbarra , @smottal ,

I addressed the multipleCheckBox error I highlighted in the earlier image as well. Please take a look at the PR again.

New implementation:
```
export function parseValue(value) {
	return Array.isArray(value) ? value.join(', ') : value;
}
```
Comma

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
@bryceosterhaus , could you take a look at the source format failure here?

If I "fix" it according to the failure being complained here, the local source format removes the empty line again.
Caused by: https://github.com/liferay/liferay-frontend-projects/pull/686

**Comment by brianikim** [comment]
@andrea-ale-sbarra , @smottal , review on this could continue while we sort out the source format failure.

**Comment by brianikim** [comment]
ci:test:sf

**Comment by holatuwol** [comment]
ci:test:sf:norebase

**Comment by brianikim** [comment]
Resent here: https://github.com/liferay-commerce/liferay-portal/pull/4326

**Pull request closed without merging**